src/main/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStepExecution.java

```diff
@@ -292,7 +292,7 @@ private static void cancel(StepContext context, Integer build) throws IOExceptio
             if (r != null) { // it should be always non-null at this point, but let's do a defensive check
                 job = r.getParent().getFullName();
             }
-            throw new FlowInterruptedException(Result.NOT_BUILT, new CancelledCause(job + "#" + build));
+            throw new FlowInterruptedException(Result.NOT_BUILT, false, new CancelledCause(job + "#" + build));
```

Comment:
You should only change this argument to `false` if this exception should be considered a non-fatal problem that would make sense for something like the `retry` step to retry. As far as I understand, that is not the case for this exception.